### PR TITLE
Remove conversion webhook from manifests

### PR DESCRIPTION
**Pull request by Jooho**
**What this PR does / why we need it**:
At the moment, kserve does not have [the conversion webhook ](https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1beta1/inference_service_conversion.go) so we don't need the conversion webhook related information from manifests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://github.com/kserve/kserve/issues/3344

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
Instead of commenting out, we can probably remove them directly from the YAML.

**Comment by Jooho** [comment]
@terrytangyuan this is a kind of conversion webhook template and afaik, there is a plan to upgrade api version near future so I think it is okay to have the reference. However, I am okay with removing it for now.

**Comment by Jooho** [comment]
@ckadner could you please retrigger the failed test?

**Comment by ckadner** [comment]
> Instead of commenting out, we can probably remove them directly from the YAML.

I would agree

@Jooho:

> there is a plan to upgrade api version near future so I think it is okay to have the reference

would you have a link to an issue or Slack discussion? or maybe KServe meeting doc?

**Comment by Jooho** [comment]
@ckadner from meeting note, https://docs.google.com/document/d/1KZUURwr9MnHXqHA08TFbfVbM8EAJSJjmaMhnvstvi-k/edit

~~~
KServe 1.0 planning
Inference service v1beta1 -> v1
Serving runtime v1alpha1 -> v1beta1
Inference Graph v1alpha1 -> v1beta1
~~~

**Comment by Jooho** [comment]
@terrytangyuan @ckadner I deleted related parts and files.

**Comment by israel-hdez** [comment]
> this is a kind of conversion webhook template and afaik, there is a plan to upgrade api version near future so I think it is okay to have the reference. However, I am okay with removing it for now.

I think we should be able to re-generate these artifacts with `kubebuilder`. So, I'm also +1 on removing.

**Review by terrytangyuan — APPROVED** [review]
/lgtm

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -76,13 +71,8 @@ replacements:
     fieldPath: metadata.namespace
     kind: Deployment
     name: kserve-controller-manager
-    version: v1
+    version: v1
```
Comment (ckadner) [inline config/default/kustomization.yaml]: remove trailing white spaces

**Review by ckadner — CHANGES_REQUESTED** [review]
Looks good -- just one tiny nit pick on trailing whitespace

File: config/default/kustomization.yaml [hunk 2]
```diff
@@ -123,14 +113,6 @@ replacements:
       kind: Certificate
       name: serving-cert
       namespace: kserve
-  - fieldPaths:
-    - metadata.annotations.[cert-manager.io/inject-ca-from]
```
Comment (yuzisun) [inline config/default/kustomization.yaml]: Should this be removed?

Comment (Jooho) [reply config/default/kustomization.yaml]: You are right. I will rollback this part and file.

File: config/default/kustomization.yaml (hunk 1 above)
Comment (Jooho) [reply config/default/kustomization.yaml]: updated

**Pull request merged**